src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java

```diff
@@ -217,6 +217,24 @@ public abstract class AbstractHpiMojo extends AbstractJenkinsMojo {
     @Parameter
     protected boolean pluginFirstClassLoader = false;
 
+    /**
+     * If {@code true}, then if {@link #bundledArtifacts} does not match the actual list of
+     * bundled artifacts, the build will fail.
+     * @since TODO
+     */
+    @Parameter(defaultValue = "${hpi.strictBundledArtifacts}")
+    private boolean strictBundledArtifacts;
```

Comment:
If anyone feels strongly about naming, e.g. library vs artifact, packaged vs bundled, whatever, feel free to provide suggestions.